vmdb/app/models/miq_provision_task_configured_system_foreman/options_helper.rb

```diff
@@ -1,21 +1,31 @@
 require 'manageiq_foreman'
 module MiqProvisionTaskConfiguredSystemForeman::OptionsHelper
   def log_provider_options
-    log_header = "MIQ(#{self.class.name}##{__method__})"
-    $log.info("#{log_header} Provisioning [#{source.name}]")
+    $log.info("MIQ(#{self.class.name}##{__method__}) Provisioning [#{source.name}]")
   end
 
   def merge_provider_options_from_automate
     phase_context[:provider_options].merge!(get_option(:provider_options) || {}).delete_nils
-    dumpObj(phase_context[:provider_options], "MIQ(#{self.class.name}##{__method__}) Merged Provider Options: ", $log, :info)
+    dumpObj(phase_context[:provider_options], "MIQ(#{self.class.name}##{__method__}) Merged Provider Options: ", $log, :info, :protected => {:path => /root_pass/})
   end
 
   def prepare_provider_options
-    phase_context[:provider_options] = {
-      "id"           => source.manager_ref,
-      "hostgroup_id" => dest_configuration_profile.manager_ref,
-    }
-    dumpObj(phase_context[:provider_options], "MIQ(#{self.class.name}##{__method__}) Default Provider Options: ", $log, :info)
+    h = {"hostgroup_id" => dest_configuration_profile.manager_ref}
+    h["name"]      = options[:hostname]                           if options[:hostname]
+    h["ip"]        = options[:ip_addr]                            if options[:ip_addr]
+    h["root_pass"] = MiqPassword.decrypt(options[:root_password]) if options[:root_password]
+    phase_context[:provider_options] = h
+    dumpObj(phase_context[:provider_options], "MIQ(#{self.class.name}##{__method__}) Default Provider Options: ", $log, :info, :protected => {:path => /root_pass/})
+  end
+
+  def update_configured_system_in_provider
+    phase_context[:provider_options].each do |k, v|
+      begin
+        source.with_provider_object { |cs| cs.update(k => v) }
```

Comment:
@brandondunne Think I'm leaning towards updating everything at once. And not due to optimization.

If we change the attributes first without changing the hostgroup, they might not match the old hostgroup. And it will fail.
If we change the hostgroup without changing the attributes, the previous attributes may not match the old attributes. And it will fail.

Setting the location/organization first will possibly clash with the hostgroup or attributes.

Almost seems like we may need to set these values to nil first, set the location, then hostgroup, then set all the values. (only setting them to nil if we are trying to change those values)

It does all seem a little like overkill.
Also seems like the foreman server will just reset these values back to the old value if it is not configured correctly.